### PR TITLE
New Rule: Link: IPFS string found in URL

### PR DESCRIPTION
**Pull request by morriscode**
- New Rule: Link: IPFS string found in link
- Updating name

### CODE REVIEW

**Review by bfilar — APPROVED** [review]

File: link_ipfs_phishing.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: "Link: IPFS string found in URL"
+description: "Detects emails containing links that have 'ipfs' in the domain, or unanalyzed links
```
Comment (jkamdjou) [inline link_ipfs_phishing.yml:2]:
```suggestion
description: "Detects messages containing links that have 'ipfs' in the domain, or unanalyzed links
```

File: link_ipfs_phishing.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  that contain 'ipfs' in the url"
```
Comment (jkamdjou) [inline link_ipfs_phishing.yml:3]:
```suggestion
  that contain 'ipfs' in the url. IPFS has been recently observed hosting phishing sites."
```

File: link_ipfs_phishing.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+references:
+  - "https://securelist.com/ipfs-phishing/109158/"
```
Comment (jkamdjou) [inline link_ipfs_phishing.yml:5]:
```suggestion
  - "https://securelist.com/ipfs-phishing/109158/"
  - "https://docs.ipfs.tech/how-to/address-ipfs-on-web/"
```

File: link_ipfs_phishing.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  
+    // Or we didn't submit the URL, and the url contains ipfs and is not common or in org_domains
+  or (beta.linkanalysis(.).submitted == false) and any(body.links, strings.icontains(.href_url.url, "ipfs")
+      and .href_url.domain.domain not in $free_file_hosts and
+          .href_url.domain.domain not in $free_subdomain_hosts and
+          .href_url.domain.domain not in $tranco_1m and
+          .href_url.domain.domain not in $umbrella_1m and
+          .href_url.domain.domain not in $org_domains
+      )
+  )
```
Comment (jkamdjou) [inline link_ipfs_phishing.yml:22]:
```suggestion
  type.inbound

  and any(body.links,
    // Any body link domains contain "ipfs"
    strings.icontains(.href_url.domain.domain, "ipfs")
  
    // Or we didn't submit the URL, and the url contains ipfs and is not common
    or (
        beta.linkanalysis(.).submitted == false
        and strings.icontains(.href_url.url, "ipfs")
        and .href_url.domain.domain not in $org_domains
        and (
            (
                // don't include high rep domains
                .href_url.domain.domain not in $tranco_1m
                and .href_url.domain.domain not in $umbrella_1m
            )
            // if it's in Tranco or Umbrella, still include it if it's one of these
            or .href_url.domain.domain in $free_file_hosts
            or .href_url.domain.root_domain in $free_subdomain_hosts
        )
      )
  )

  // unsolicited
  and (
        (
            sender.email.domain.root_domain in $free_email_providers
            and sender.email.email not in $recipient_emails
        )
        or (
            sender.email.domain.root_domain not in $free_email_providers
            and sender.email.domain.domain not in $recipient_domains
        )
  )
```

Comment (jkamdjou) [reply link_ipfs_phishing.yml:22]: you had a double loop of `body.links` in there. also i think this might be the domain exclusion logic you intended?

**Pull request merged**